graphql_compiler/schema_generation/sqlalchemy/schema_graph_builder.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import OrderedDict
+import warnings
+
+from graphql.type import GraphQLBoolean, GraphQLFloat, GraphQLInt, GraphQLString
+import six
+import sqlalchemy.sql.sqltypes as sqltypes
+
+from ...schema import GraphQLDate, GraphQLDateTime, GraphQLDecimal
+from ..schema_graph import PropertyDescriptor, SchemaGraph, VertexType
+
+
+# TODO(pmantica1): Add scalar mapping for the following classes: Interval, and Time.
+# We do not currently plan to add a mapping for JSON and Binary objects.
+SQL_CLASS_TO_GRAPHQL_TYPE = OrderedDict([
+    (sqltypes.String, GraphQLString),
+    (sqltypes.Integer, GraphQLInt),
+    (sqltypes.Float, GraphQLFloat),
+    (sqltypes.Numeric, GraphQLDecimal),
+    (sqltypes.DateTime, GraphQLDateTime),
+    (sqltypes.Date, GraphQLDate),
+    (sqltypes.Boolean, GraphQLBoolean),
+])
+
+
+# TODO(pmantica1): Map foreign keys to edges.
+# TODO(pmantica1): Represent table inheritance in SchemaGraph.
+# TODO(pmantica1): Add option to map tables to EdgeTypes instead of VertexTypes.
+# TODO(pmantica1): Map arrays to GraphQLLists.
+def get_schema_graph_from_sql_alchemy_metadata(sqlalchemy_metadata):
+    """Return the matching SchemaGraph for the SQLAlchemy Metadata object"""
+    _validate_sql_to_graphql_is_toposorted_by_class_inheritance()
```

Comment:
Since this is verifying that a global constant is valid, why not make it a test case instead of validating it on every call of this function?

Reply:
Good idea.

Reply:
There is no longer any need for the dict to be toposorted.